### PR TITLE
Problem Suggestion: Add "subarray squares" to Platinum - Divide & Conquer - DP

### LINKED ISSUE #1099: Problem Suggestion: Add "subarray squares" to Platinum - Divide & Conquer - DP
*Ray Tang* (UID jjkJgktEM3brQnXtcoJaqtkjnYg1) suggested adding the problem [subarray squares](https://cses.fi/problemset/task/2086) to the module [Platinum - Divide & Conquer - DP](https://usaco.guide/plat/DC-DP#problemlist-dnc_general).

**Difficulty**: Easy
**Tags**: 
**Additional Notes**: None

*This report was automatically generated from a user submitted problem suggestion on the USACO guide.*

### PR DESCRIPTION
**Pull request by jeffkmeng**
*Problem Suggestion Issue Migration Bot* (UID PROB_SUGG_MIGRATE_BOT) suggested adding the problem [subarray squares](https://cses.fi/problemset/task/2086) to the `dnc_general` table of the module [Platinum - Divide & Conquer - DP](https://usaco.guide/plat/DC-DP#problemlist-dnc_general).

**Automatically Generated JSON:**
```json
{
  "uniqueId": "cses-2086",
  "name": "subarray squares",
  "url": "https://cses.fi/problemset/task/2086",
  "source": "CSES",
  "difficulty": "Easy",
  "isStarred": false,
  "tags": [
    ""
  ],
  "solutionMetadata": {
    "kind": "none"
  }
}
```
**Additional Notes**:
None
(Closes #1099)

*This PR was automatically generated from a user-submitted problem suggestion on the USACO guide.*

### CODE REVIEW

File: content/5_Plat/DC-DP.problems.json [hunk 1]
```diff
@@ -15,6 +15,18 @@
...
   ],
   "dnc_general": [
+    {
+      "uniqueId": "cses-2086",
+      "name": "subarray squares",
+      "url": "https://cses.fi/problemset/task/2086",
+      "source": "CSES",
+      "difficulty": "Easy",
+      "isStarred": false,
+      "tags": [""],
```
Comment (dongliu0426) [inline content/5_Plat/DC-DP.problems.json:25]:
```suggestion
      "tags": [],
```

**Review by dongliu0426 — APPROVED** [review]

File: content/5_Plat/DC-DP.problems.json [hunk 2]
```diff
@@ -15,6 +15,18 @@
     }
   ],
   "dnc_general": [
+    {
+      "uniqueId": "cses-2086",
+      "name": "subarray squares",
```
Comment (dongliu0426) [inline content/5_Plat/DC-DP.problems.json:20]:
```suggestion
      "name": "Subarray Squares",
```

**Review by bqi343 — CHANGES_REQUESTED** [review]
move to convex hull trick

note - this is somewhat similar to circular barn

**Comment by andrewwangva** [comment]
I agree with ben this is very similar to cbarn - but also how do we differentiate between cht/d&c if they can be solved w both?

**Comment by bqi343** [comment]
cht removes a log factor from the complexity

**Review by bqi343 — CHANGES_REQUESTED** [review]
this is essentially the same as APIO "split the sequence" - do we want to keep both?

**Pull request closed without merging**